model/jpa/src/main/resources/META-INF/jpa-changelog-26.5.0.xml

```diff
@@ -0,0 +1,55 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<!--
+  ~ * Copyright 2025 Red Hat, Inc. and/or its affiliates
+  ~ * and other contributors as indicated by the @author tags.
+  ~ *
+  ~ * Licensed under the Apache License, Version 2.0 (the "License");
+  ~ * you may not use this file except in compliance with the License.
+  ~ * You may obtain a copy of the License at
+  ~ *
+  ~ * http://www.apache.org/licenses/LICENSE-2.0
+  ~ *
+  ~ * Unless required by applicable law or agreed to in writing, software
+  ~ * distributed under the License is distributed on an "AS IS" BASIS,
+  ~ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ * See the License for the specific language governing permissions and
+  ~ * limitations under the License.
+  -->
+<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.1.xsd">
+
+    <!-- These indexes IDX_OFFLINE_CSS_BY_CLIENT and IDX_OFFLINE_CSS_BY_CLIENT_STORAGE_PROVIDER are used to retrieve
+    sessions by client and to count sessions by client. They are also used to remove client sessions for no-longer
+    existing clients from the table. -->
+
+    <changeSet author="keycloak" id="26.5.0-index-offline-css-by-client">
+        <preConditions onFail="MARK_RAN" onSqlOutput="TEST">
+            <indexExists tableName="OFFLINE_CLIENT_SESSION" indexName="IDX_OFFLINE_CSS_BY_CLIENT" />
+        </preConditions>
+        <createIndex tableName="OFFLINE_CLIENT_SESSION" indexName="IDX_OFFLINE_CSS_BY_CLIENT">
+            <column name="CLIENT_ID" type="VARCHAR(255)" />
+            <column name="OFFLINE_FLAG" type="VARCHAR(4)"/>
+        </createIndex>
+    </changeSet>
+
+    <changeSet author="keycloak" id="26.5.0-index-offline-css-by-client-storage-provider">
+        <preConditions onFail="MARK_RAN" onSqlOutput="TEST">
+            <indexExists tableName="OFFLINE_CLIENT_SESSION" indexName="IDX_OFFLINE_CSS_BY_CLIENT_STORAGE_PROVIDER" />
+        </preConditions>
+        <createIndex tableName="OFFLINE_CLIENT_SESSION" indexName="IDX_OFFLINE_CSS_BY_CLIENT_STORAGE_PROVIDER">
+            <column name="CLIENT_STORAGE_PROVIDER" type="VARCHAR(36)" />
+            <column name="EXTERNAL_CLIENT_ID" type="VARCHAR(255)"/>
+            <column name="OFFLINE_FLAG" type="VARCHAR(4)"/>
+        </createIndex>
+    </changeSet>
```

Comment:
A client session would be either for an external client, or an internal client, but never both. So I'd want to combine those two index into one, which then is `CLIENT_ID`, `CLIENT_STORAGE_PROVIDER`, `EXTERNAL_CLIENT_ID`, `OFFLINE_FLAG`. 

Whenever we then select an external client, we would also pass in `CLIENT_ID` with the value `external`. 

I would hope it would reduce the indexing overhead. WDYT?

Reply:
Transferring your concern from https://github.com/keycloak/keycloak/pull/43566#discussion_r2449176528 here:
> now realize that this would be difficult due to the columns that are `NULL`, and I am pretty sure that different databases would handle it differently. So it might then be two indexes after all ... on PostgreSQL, I would love to use a partial index instead. Unfortunately Liquibase doesn't support that.

Reply:
Leaving e4486667d439b2ca0fb11eb196485ccd279634d4 and ca58c2921eeef1c6b19e8ea879a47808c8137dd0 in the history for now and remove them once we have agreed